dissect/volume/vinum/vinum.py

```diff
@@ -0,0 +1,340 @@
+# For more information see:
+# https://docs.freebsd.org/en/articles/vinum/
+# https://www.usenix.org/legacy/events/usenix99/full_papers/lehey/lehey.pdf
+from __future__ import annotations
+
+import io
+import logging
+import os
+from collections import defaultdict
+from functools import cached_property
+from typing import TYPE_CHECKING, BinaryIO, TypedDict, TypeVar
+
+from dissect.util import ts
+
+from dissect.volume.raid.raid import (
+    RAID,
+    Configuration,
+    DiskMap,
+    PhysicalDisk,
+    VirtualDisk,
+)
+from dissect.volume.raid.stream import Layout, Level
+from dissect.volume.vinum.c_vinum import MAGIC_ACTIVE, MAGIC_INACTIVE, c_vinum
+from dissect.volume.vinum.config import (
+    SD,
+    Plex,
+    PlexOrg,
+    PlexState,
+    SDState,
+    Volume,
+    parse_vinum_config,
+)
+
+if TYPE_CHECKING:
+    VinumPhysicalDiskDescriptor = BinaryIO | "VinumPhysicalDisk"
+
+log = logging.getLogger(__name__)
+log.setLevel(os.getenv("DISSECT_LOG_VINUM", "CRITICAL"))
+log.addHandler(logging.StreamHandler())
```

Comment:
```suggestion
```
We don't do this currently in other places, what does it exactly do?

Reply:
Don't know how that ended up here. The idea was to use the same construct we use in the other volume implementations. I'll remove it.